flink-runtime/src/test/java/org/apache/flink/runtime/io/network/api/serialization/SpanningRecordSerializerTest.java

```diff
@@ -169,25 +167,26 @@ private void test(Util.MockRecords records, int segmentSize) throws Exception {
 
 		// -------------------------------------------------------------------------------------------------------------
 
-		serializer.continueWritingWithNextBufferBuilder(createBufferBuilder(segmentSize));
-
+		BufferBuilder bufferBuilder = createBufferBuilder(segmentSize);
 		int numBytes = 0;
 		for (SerializationTestType record : records) {
-			RecordSerializer.SerializationResult result = serializer.addRecord(record);
+			serializer.serializeRecord(record);
+			RecordSerializer.SerializationResult result = serializer.copyToBufferBuilder(bufferBuilder);
 			numBytes += record.length() + serializationOverhead;
 
 			if (numBytes < segmentSize) {
 				Assert.assertEquals(RecordSerializer.SerializationResult.FULL_RECORD, result);
 			} else if (numBytes == segmentSize) {
 				Assert.assertEquals(RecordSerializer.SerializationResult.FULL_RECORD_MEMORY_SEGMENT_FULL, result);
-				serializer.continueWritingWithNextBufferBuilder(createBufferBuilder(segmentSize));
+				bufferBuilder = createBufferBuilder(segmentSize);
 				numBytes = 0;
 			} else {
 				Assert.assertEquals(RecordSerializer.SerializationResult.PARTIAL_RECORD_MEMORY_SEGMENT_FULL, result);
 
 				while (result.isFullBuffer()) {
 					numBytes -= segmentSize;
-					result = serializer.continueWritingWithNextBufferBuilder(createBufferBuilder(segmentSize));
+					bufferBuilder = createBufferBuilder(segmentSize);
+					result = serializer.copyToBufferBuilder(bufferBuilder);
```

Comment:
I know, this wasn't checked before, but should we actually also check for a full record after this `while` loop?